drivers/periph_common/flashpage.c

```diff
@@ -104,4 +106,79 @@ int flashpage_rwwee_write_and_verify(unsigned page, const void *data)
 
 #endif
 
+/* We have to write whole words, but writing 0xFF is basically a no-op
+ * so fill the unaligned bytes with 0xFF to get a whole extra word.
+ * To support writes of data with less than 4 bytes, an offset needs
+ * to be supplied.
+ */
+static void unaligned_pad_start(const void *dst, uint8_t *buffer,
+                                const void *data, uint8_t len, uint8_t offset)
+{
+    if (IS_ACTIVE(FLASHPAGE_NO_BLIND_WRITES)) {
+        memcpy(buffer, dst, FLASHPAGE_WRITE_BLOCK_SIZE);
+    } else {
+        memset(buffer, FLASHPAGE_ERASE_STATE, FLASHPAGE_WRITE_BLOCK_SIZE);
+    }
+    memcpy(buffer + offset, data, len);
+}
+
+/* We have to write whole words, but writing 0xFF is basically a no-op
+ * so fill the unaligned bytes with 0xFF to get a whole extra word.
+ */
+static void unaligned_pad_end(const void *dst, uint8_t *buffer,
+                              const void *data, uint8_t len)
+{
+    if (IS_ACTIVE(FLASHPAGE_NO_BLIND_WRITES)) {
+        memcpy(buffer, dst, FLASHPAGE_WRITE_BLOCK_SIZE);
+    } else {
+        memset(buffer, FLASHPAGE_ERASE_STATE, FLASHPAGE_WRITE_BLOCK_SIZE);
+    }
+    memcpy(buffer, data, len);
+}
+
+void flashpage_write_unaligned(void *_dst, const void *_data, size_t len)
+{
+    uint8_t *dst = _dst;
+    const uint8_t *data = _data;
+
+    const uint8_t mask = FLASHPAGE_WRITE_BLOCK_ALIGNMENT - 1;
+    uint8_t padding[FLASHPAGE_WRITE_BLOCK_SIZE];
```

Comment:
This can be as big as 256 bytes for the lpc23xx. For all other MCUs this seems to be something that should easily fit on the stack, though.

Maybe add a warning to the documentation that stack allocation can be large for strange MCUs?